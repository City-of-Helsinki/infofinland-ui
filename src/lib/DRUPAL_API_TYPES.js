export const NODE_TYPES = {
  PAGE: 'node--page',
  LANDING_PAGE: 'node--landing_page',
  PVT_NODE: 'node--office_contact_info',
}

export const CONTENT_TYPES = {
  TEXT: 'paragraph--text',
  HEADING: 'paragraph--heading',
  PARAGRAPH_IMAGE: 'paragraph--image',
  ACCORDION: 'paragraph--accordion',
  ACCORDION_ITEM: 'paragraph--accordion_item',
  HERO: 'paragraph--hero',
  READMORE: 'paragraph--language_link_collection',
  READMORE_LINK_COLLECTION: 'node--link',
  READMORE_LINK: 'paragraph--language_link',
  VIDEO_REMOTE: 'paragraph--remote_video',
  VIDEO_HELSINKI: 'paragraph--helsinki_kanava',
  LOCALINFO: 'paragraph--municipality_liftup',
  FILE: 'file--file',
  PTV: 'paragraph--ptv_contact',
  PTV_NODE: 'node--office_contact_info',
  MEDIA_IMAGE: 'media--image',
  MEDIA_VIDEO: 'media--remote_video',
  COLUMNS: 'paragraph--columns',
  COLUMN_LEFT: 'field_columns_left_column',
  COLUMN_RIGHT: 'field_columns_right_column',
  LANGUAGE: 'taxonomy_term--language',
  MUNICIPALITY: 'taxonomy_term--municipalitys',
  HERO_COLOR: 'taxonomy_term--colors',
}

export const TEXT_HTML_FORMAT = 'full_html'
