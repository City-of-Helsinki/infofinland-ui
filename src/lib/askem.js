// Set third party script tag and resolve promise after the script is ready to use.
// On resolve, return the reset function
// Reject after a couple of seconds.
const loadAskemScript = () => {
  return new Promise((resolve, reject) => {
    if (!document.getElementById('if-askem-script')) {
      document.body.append(getAskemScriptTag())
    }

    let i = 1;
    const max = 3;

    const intervalId = setInterval(()=> {
      if (window.askem.reset) {
        resolve(window.askem.reset);
        clearInterval(intervalId);
      }
      if (i > max) {
        reject(false);
        clearInterval(intervalId);
      }
      i++;
    }, 750);
  })
}

const removeAskemScript = () => {
  document.body.removeChild(getAskemScriptTag());
}

const getAskemScriptTag = () => {
  const script = document.createElement('script');
  script.src = 'https://cdn.askem.com/plugin/askem.js';
  script.type = 'text/javascript';
  script.id = 'if-askem-script';

  return script;
}

export {loadAskemScript, getAskemScriptTag, removeAskemScript};
