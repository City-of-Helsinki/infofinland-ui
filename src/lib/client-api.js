import axios from 'axios'

/** The Client API urls  */
export const SEARCH_URL = '/api/search/'
export const LOCAL_INFO_URL = '/api/localinfo'
export const LOCALES_URL = '/api/available-locales'

/**
 * Get search results based on searchQueryValue
 * @param {string} search - the search term.
 */

export const getSearchResults = async ({ search, locale }) => {
  const { data } = await axios(`${SEARCH_URL}${locale}`, { params: { search } })
  return data
}

export const getLocalInformation = async (params) => {
  const { data } = await axios.get(LOCAL_INFO_URL, { params })
  return data?.node
}

/**
 * get list of localized pages for given path
 * @param {string} search - the search term.
 */

export const getLocalesForPath = async ({ path }) => {
  const { data } = await axios.get(LOCALES_URL, { params: { path } })
  return data
}
