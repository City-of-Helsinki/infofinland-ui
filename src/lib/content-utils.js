export const getFit = ({ width, height }) =>
  width < height ? 'contain' : 'cover'
// But my gosh donchu know it...
