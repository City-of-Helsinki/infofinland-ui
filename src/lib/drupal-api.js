import {
  getMenu,
  // getPathsFromContext,
  // getResourceFromContext,
  getResource,
} from 'next-drupal'

export const getMainMenu = async (context) => await getMenu('main', context)

// export  getLocalInformationBlock = async (pageId) => await getResource('','')

export const getFooterAboutMenu = async (context) =>
  await getMenu('footer-about', context)
export const getAboutMenu = async (context) => await getMenu('about', context)

// export const getPage = async () => await getMenu('main')

export const getFeedbackInstructionsBlock = async () => await getResource()

// "block_content--basic",
// "97ee0b84-309f-4b5e-b321-042018552428"

export const getCommonApiContent = async (context) => {
  const [mainMenu, aboutMenu] = await Promise.all([
    getMainMenu(context),
    getFooterAboutMenu(context),
  ])
  return { mainMenu, aboutMenu }
}
