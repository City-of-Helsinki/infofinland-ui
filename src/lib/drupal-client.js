import { DrupalClient } from 'next-drupal'
import getConfig from 'next/config'
export const getDrupalClient = (withAuth) => {
  const {
    NEXT_PUBLIC_DRUPAL_BASE_URL,
    DRUPAL_PREVIEW_SECRET,
    DRUPAL_CLIENT_ID,
    DRUPAL_CLIENT_SECRET,
    REVALIDATE_TIME,
    DEBUG,
  } = getConfig().serverRuntimeConfig
  return new DrupalClient(NEXT_PUBLIC_DRUPAL_BASE_URL, {
    auth: {
      clientId: DRUPAL_CLIENT_ID,
      clientSecret: DRUPAL_CLIENT_SECRET,
    },
    //cache: redisCache,
    previewSecret: DRUPAL_PREVIEW_SECRET,
    // Make preview work in development environment.
    forceIframeSameSiteCookie: process.env.NODE_ENV === 'development',
    ...(withAuth && { withAuth: true }),
    next: { revalidate: REVALIDATE_TIME },
    debug: DEBUG
  })
}

export default getDrupalClient
