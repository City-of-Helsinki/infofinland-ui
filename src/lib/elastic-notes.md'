CERT TESTING


- IP:port , nocerts, user/pw  -> error: self signed cert

- IP:port , ssl- public-cert, user/pw  -> ??
