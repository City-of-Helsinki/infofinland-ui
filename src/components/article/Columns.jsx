import Block from "../layout/Block"
import { ArticleImage } from "./ImageBlock"
import ParseHtml from "../ParseHtml"
import { CONTENT_TYPES } from "@/lib/DRUPAL_API_TYPES"
import { TEXT_HTML_FORMAT } from "./ContentMapper"
import cls from 'classnames'
import { getImage } from "@/lib/ssr-api"

const ColumnMapper = ({type,field_text,field_image, field_image_caption})=> {

  switch (type) {
    case CONTENT_TYPES.TEXT:
      if (
        field_text?.format !== TEXT_HTML_FORMAT ||
        !field_text?.processed
      ) {
        return null
      }
      return (
      <div className={cls('ifu-article__bodyblock', field_text?.format)}>
        <ParseHtml html={field_text?.processed} />
      </div>)

    case CONTENT_TYPES.PARAGRAPH_IMAGE:
      return <ArticleImage {...getImage({field_image})} caption={field_image_caption}/>

    default:
      return null

  }
}

export default function Columns({field_columns_left_column, field_columns_right_column})  {
  return (
    <Block>
      <div className="lg:grid grid-cols-2 gap-6">
        <div className="mb-8 lg:mb-0">
          <ColumnMapper {...field_columns_left_column}/>
        </div>
        <div className="mb-8 lg:mb-0">
          <ColumnMapper {...field_columns_right_column}/>
        </div>
      </div>
    </Block>
  )
}
