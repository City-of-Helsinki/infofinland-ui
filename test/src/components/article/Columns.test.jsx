/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import Columns from '@/components/article/Columns'
import { CONTENT_TYPES, TEXT_HTML_FORMAT } from '@/lib/DRUPAL_API_TYPES'

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (i18nKey) => i18nKey,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

const CONTENT =
  'Alkukartoituksessa arvioidaan, mitkä palvelut auttavat sinua kotoutumaan.'

describe('Columns', () => {
  it('should render text and images', () => {
    const { asFragment } = render(
      <Columns
        field_columns_left_column={{
          type: CONTENT_TYPES.TEXT,
          field_text: { processed: CONTENT, format: TEXT_HTML_FORMAT },
        }}
        field_columns_right_column={{
          type: CONTENT_TYPES.PARAGRAPH_IMAGE,
          field_image: {
            field_image_caption: 'Kuvatekstiä',
            field_media_image: {
              resourceObjectMeta: {
                alt: 'alt',
                title: 'title',
                height: 200,
                width: 600,
              },
              uri: { url: '/FOO.png' },
            },
          },
        }}
      />
    )
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
