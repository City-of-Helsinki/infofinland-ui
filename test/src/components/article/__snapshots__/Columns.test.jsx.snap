// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns should render text and images 1`] = `
<DocumentFragment>
  <div
    class="px-6 lg:px-12 lg:mx-12  xl:mx-28 2xl:mx-48  3xl:ms-64  3xl:max-w-4xl py-16 ifu-article__columns"
  >
    <div
      class="lg:grid grid-cols-2 gap-16"
    >
      <div
        class="mb-8 lg:mb-0"
      >
        <div
          class="ifu-article__column-block full_html"
        >
          Alkukartoituksessa arvioidaan, mitkä palvelut auttavat sinua kotoutumaan.
        </div>
      </div>
      <div
        class="mb-8 lg:mb-0"
      >
        <div
          class="overflow-hidden relative rounded aspect-square"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
            />
            <noscript />
          </span>
        </div>
        <p
          class="mt-2 text-body-small text-gray-dark"
        >
          Kuvatekstiä
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
